### ISSUE TITLE
The sound effect handling in Opentct2 is a bit odd.

### ISSUE THREAD

**Issue by guljam** [OpenRCT2/OpenRCT2#17285]
### Operating System

Windows 11 64bit

### OpenRCT2 build

OpenRCT2, v0.4.0 (d5d52f4f on develop) provided by Github

### Describe the issue

It's weird compared to vanilla RCT2 when certain audio effects play quickly.

vanilla RCT2 audio behavior
https://youtu.be/dhzNsvkWy-U 

Openrct2 audio behavior
https://youtu.be/12-jSmK0450 

In my opinion, the audio processing of openrct2 seems to play the next sound effect after the playback of a specific sound effect is completely completed.

As for the sound effect of vanilla RCT2, when a specific sound effect is playing and the same sound effect is played, the previous sound effect is stopped and a new sound effect is played.

and another issue

It's a very, very small difference
The "PlaceItem" sound effect is a bit quieter than the vanilla RCT2 sound effect.
This is to the point where you won't notice a difference if you're not focused.

I tried to solve this problem myself
It is impossible because there is no knowledge about audio.
Report the issue.

### Steps to reproduce

1. 
2. 
3.

**Comment by IntelOrca** [comment]
Not sure about the first issue, but the second is most likely a duplicate of #7490

**Comment by guljam** [comment]
@IntelOrca yes, with this patch, the button's sound volume is clearly audible, but there is a very subtle volume difference.

**Comment by guljam** [comment]
@IntelOrca The first issue seems to be related to the **AudioMixer::GetNextAudioChunk** function.
This function seems to be update much slower than 1tick in the game. (every 0.2 seconds in my opinion)
So, if a large amount of sound effects occur within 0.2 seconds, they are accumulated in the _channels list and sound effects are played all at once.

**Comment by IntelOrca** [comment]
@guljam I might postpone that fix to when we switch to using OpenAL.
